### PR TITLE
Dropdown: Fix keyboard accessibility

### LINKED ISSUE #84652: [Accessibility issue] Dropdown keyboard accessibility regression with up / down keys
**Steps to reproduce**:
Go to any instance of Grafana (such as play.grafana.com) and tab to a dropdown menu and hit space /  enter. Use the up / down arrows to navigate the options. Notice nothing happens.

**Actual Result**
Nothing happens (sometimes it works correctly the first time on a fresh window but it consistently never works in any subsequent tries).

**Expected Result**
I would expect focus to move to the first menu item automatically on toggle of the dropdown opening and the up / down arrows to work correctly.

Currently you have to tab to the menu itself (which has a tabIndex of -1 and it has no focus indicator...) before you can begin using the up / down keys. If you use a mouse to click the toggle then the up / down keys work immediately.

On that note, if you have focus on the dropdown toggle and then hover it with your mouse it loses its focus styles.

It also means that after opening the menu you have to tab multiple times to get to the next element as you tab back to the dropdown toggle, rather than skipping to the next element.

Notice in this video of the behaviour it works as expected the first time round  but not subsequent times.

https://github.com/grafana/grafana/assets/6906380/3ab9ae06-96b9-4a79-b694-303b9493efb7

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- correctly passes `isMenuOpen` to `useMenuFocus`
- removes the `openedWithArrow`/`setOpenedWithArrow` concepts as we want the same behaviour no matter what
  - ~technically a breaking change for `Menu`?~
    - nope, both `MenuItem` and `SubMenu` are marked as `@internal` 🙌 
  - i suspect no one is using these properties externally as they're very internal 
- replaces `FocusScope` with `FloatingFocusManager` (this doesn't really change anything, just thought we might as well while we're here)

**Why do we need this feature?**

- so it's properly keyboard accessible
- think this is actually an overall improvement and not just fixing a regression, as before the focus would be on the containing menu element and you'd have to press the down arrow before focusing an individual option. now we immediately focus the first option in the menu 🥳 

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes #84652 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -66,17 +66,17 @@ export const Menu = Object.assign(MenuComp, {
...
-      background: `${theme.colors.background.primary}`,
-      boxShadow: `${theme.shadows.z3}`,
-      display: `inline-block`,
-      borderRadius: `${theme.shape.radius.default}`,
-      padding: `${theme.spacing(0.5, 0)}`,
+      background: theme.colors.background.primary,
+      boxShadow: theme.shadows.z3,
+      display: 'inline-block',
+      borderRadius: theme.shape.radius.default,
+      padding: theme.spacing(0.5, 0),
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Menu/Menu.tsx:79]: this also isn't really doing anything, it's just removing the unnecessary template string wrapping

**Review by ckbedwell — CHANGES_REQUESTED** [review]
Amazing turnaround, thank you for taking a look at this so quickly 🥳 

A few comments on using it as a user as not familiar enough with the floating-ui api yet to comment on the code (despite the lovely walkthrough today at the bi-weekly):

- I think if the dropdown is open and you tab again it should just take you to the next item rather than restore focus to the toggle (it should restore if you shift-tab though (which is what happens).

I've just been looking at different dropdowns across different sites I like for accessibility and that is the pattern they follow -- it is also the pattern they have on floating-ui.com when looking at their dropdown menu.

- When I use my mouse and click on a dropdown the selected state doesn't appear for the first item (so in this case the background highlight)  despite it is selected, is it worth taking a look at that with this PR, too?

File: packages/grafana-ui/src/components/Menu/Menu.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Menu/Menu.tsx:79]: I think we could use `Box` and get rid of these styles altogether.

**Comment by ashharrison90** [comment]
> Amazing turnaround, thank you for taking a look at this so quickly 🥳
> 
> A few comments on using it as a user as not familiar enough with the floating-ui api yet to comment on the code (despite the lovely walkthrough today at the bi-weekly):
> 
> * I think if the dropdown is open and you tab again it should just take you to the next item rather than restore focus to the toggle (it should restore if you shift-tab though (which is what happens).
> 
> I've just been looking at different dropdowns across different sites I like for accessibility and that is the pattern they follow -- it is also the pattern they have on floating-ui.com when looking at their dropdown menu.
> 
> * When I use my mouse and click on a dropdown the selected state doesn't appear for the first item (so in this case the background highlight)  despite it is selected, is it worth taking a look at that with this PR, too?

@ckbedwell 

the problem is this doesn't _really_ use much of the floating-ui api. there's a custom hook [here](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Menu/hooks.ts#L23) that actually handles most of the keyboard interactions. i can't quite figure out how to get the behaviour you want. i do agree though.

re: showing the focused element when using the mouse. if we do that, we can end up in this situation...
image

here `View settings` is focused, but `Share` is hovered.

**Comment by ckbedwell** [comment]
> the problem is this doesn't really use much of the floating-ui api. t[here](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Menu/hooks.ts#L23)'s a custom hook here that actually handles most of the keyboard interactions. i can't quite figure out how to get the behaviour you want. i do agree though.

Okay, no problem. We can take a more detailed look at that in the future as this is a great step anyway.

> re: showing the focused element when using the mouse. if we do that, we can end up in this situation... image
> 
> here `View settings` is focused, but `Share` is hovered.

@staton-hyse11 -- I think we might be talking about this dropdown component tomorrow for another issue but I think this is related -- definitely something to think about: https://github.com/grafana/grafana/issues/79573

**Review by ckbedwell — APPROVED** [review]
LGTM 🌟

**Review by joshhunt — APPROVED** [review]
Gave this a a kick around and it seems to work well for me :)

**Pull request merged**